plugins/@grouparoo/pipedrive/src/lib/parallelism.ts

```diff
@@ -0,0 +1,24 @@
+import { AppParallelismMethod, objectCache } from "@grouparoo/core";
+
+const DEFAULT_VALUE = 10;
+
+export const parallelism: AppParallelismMethod = async ({
+  app,
+  appOptions,
+}) => {
+  const cachedValue = await objectCache(
+    {
+      objectId: app.id,
+      cacheKey: ["pipedrive:parallelism", appOptions],
+      cacheDurationMs: 2000, // 2 seconds
+      write: false, // Read-only
+    },
+    async () => null
+  );
+
+  if (cachedValue === null) {
+    return DEFAULT_VALUE;
+  }
+
+  return parseInt(cachedValue);
```

Comment:
So this will be the number of remaining requests in our period. I think we want to divide that my the number of requests that the `exportRecord` method will do.

1. findPersonIdByEmail
2. delete/update/create
3. Maybe a few to handle group creation

So let's divide by `3` here to be on the safe side.

Reply:
And floor it.
`return Math.floor(parseInt(cachedValue) / 3.0)` for safety